pennylane/devices/tests/test_measurements.py

```diff
@@ -1615,9 +1615,10 @@ def circuit():
     def test_method_overriden_by_device(self, device):
         """Test that the device can override a measurement process."""
         dev = device(2)
+
         _skip_test_for_braket(dev)
 
-        @qml.qnode(dev)
+        @qml.qnode(dev, interface="autograd")
```

Comment:
Here defining the interface is needed because `update_gradient_fn` in `__init__` is updating the device, as with `auto` this step is kipped in `__init__`. Then we apply the changes on the device before it is transformed and therefore loose the custom method.

Reply:
If you use backprop you need the interface, because `_update_gradient_fn` will potentially change the device. I think it will be easier to adress after the device revamp.

Reply:
that makes sense, I'm just worried that someone will not specify this because they didn't think they needed to? idk

Reply:
Yes I agree it is not ideal, we can add a warning if the device is changed during `__call__`. It seems to be a good compromise?

Reply:
tbh I don't think a user would hit it anyway... we can cross that bug-bridge when we get there. Largely because I'm not sure how I'd warn. save a hash of the measurement map (or the whole device?) at the end of qnode init, validate that the hash is the same at the beginning of qnode call?

Reply:
I was speaking not specifically for this test but more in general raise a warning if the device is switched in the `__call__`

Reply:
I see. I think I was looking for something a bit more general, but you're proposing to raise a warning [here](https://github.com/PennyLaneAI/pennylane/blob/876bac85c1b84c00368b7e7d34ecfcbd8b9c59b3/pennylane/qnode.py#L653-L659)?

Reply:
I added the warning and a test (test_qnode.py) let me know what you think!